File: internal/content/testdata/bench/csharp/nc-balanced-tree.cs
public class Solution {
    public bool is_balanced(TreeNode root) {
        return Height(root) >= 0;
    }
    private long Height(TreeNode node) {
        if (node == null) return 0;
        long l = Height(node.left);
        if (l < 0) return -1;
        long r = Height(node.right);
        if (r < 0) return -1;
        if (Math.Abs(l - r) > 1) return -1;
        return 1 + Math.Max(l, r);
    }
}

File: internal/content/testdata/bench/javascript/nc-balanced-tree.js
function is_balanced(root) {
    function height(node) {
        if (node === null) return 0;
        const l = height(node.left);
        if (l < 0) return -1;
        const r = height(node.right);
        if (r < 0) return -1;
        if (Math.abs(l - r) > 1) return -1;
        return 1 + Math.max(l, r);
    }
    return height(root) >= 0;
}
